### PR TITLE
Fix #2244 reset controls now skip correctly some controls

### LINKED ISSUE #2244: resetcontrols functionality always sets all the controls to false without considering the skip parameter
We need to fix this the controls.js reducer.

### PR DESCRIPTION
**Pull request by MV88**
## Description
Now reset controls rest correctly the controls. previously it was ignorign the "skip" param.

## Issues
 - Fix #2244

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)

 - [x] Bugfix

**What is the current behavior?** (You can also link to an open issue here)
controls are always reset to a false value and if we cannot avoid to reset one or more controls.

**What is the new behavior?**
now one or more controls can be skipped from resetting.

**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [x] No

### CODE REVIEW

File: web/client/reducers/__tests__/controls-test.js [hunk 1]
```diff
@@ -58,21 +68,60 @@ describe('Test the constrols reducer', () => {
...
         expect(state.c3.enabled).toNotExist();
+        expect(state.c3.idonthaveenabledfield).toExist();
+        expect(state.c3.idonthaveenabledfield).toBe("whatever");
+    });
+
+    it('reset the controls with skip c1,c3', () => {
+        const state = controls(
+            {
+                c1: { enabled: true},
+                c2: { enabled: false},
```
Comment (mbarto) [inline web/client/reducers/__tests__/controls-test.js]: The test would be better with true here

File: web/client/reducers/controls.js [hunk 2]
```diff
@@ -67,7 +67,11 @@ function controls(state = {}, action) {
             })
         });
     case RESET_CONTROLS: {
-        const resetted = Object.keys(state).reduce((previous, controlName) => {
+        let newControls = Object.keys(state);
```
Comment (mbarto) [inline web/client/reducers/controls.js]: There is always a way to use const:
const newControls = Object.keys(state).filter(c => (action.skip || []).indexOf(c) === -1)

Comment (MV88) [reply web/client/reducers/controls.js]: ok...

File: web/client/reducers/__tests__/controls-test.js (hunk 1 above)
Comment (MV88) [reply web/client/reducers/__tests__/controls-test.js]: right

**Pull request merged**